CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKHitTester.java

```diff
@@ -134,11 +354,111 @@ public int DisplayUpdate() {
+            renderPassInfo.renderArea(renderArea);       
+            renderPassInfo.pClearValues(clearValues);
+            renderPassInfo.framebuffer(vkFrameBufferHandle);
+
+            //  TODO - VK_SUBPASS_CONTENTS_INLINE  OR VK_SUBPASS_CONTENTS_SECONDARY_COMMAND_BUFFERS
+            vkCmdBeginRenderPass(commandBuffer.GetVKCommandBuffer(), renderPassInfo, VK_SUBPASS_CONTENTS_INLINE);
+            //commandBuffer.beginRenderPassCmd();
+
+            // Inheritance info for the secondary command buffers (same for all!)
+            VkCommandBufferInheritanceInfo inheritanceInfo = VkCommandBufferInheritanceInfo.callocStack(stack);
+            inheritanceInfo.sType(VK_STRUCTURE_TYPE_COMMAND_BUFFER_INHERITANCE_INFO);
+            inheritanceInfo.pNext(0);
+            inheritanceInfo.framebuffer(vkFrameBufferHandle);
+            inheritanceInfo.renderPass(cvkSwapChain.GetOffscreenRenderPassHandle());
+            //inheritanceInfo.subpass(0); // TODO Get the subpass or make it here?
+            inheritanceInfo.occlusionQueryEnable(false);
+            inheritanceInfo.queryFlags(0);
+            inheritanceInfo.pipelineStatistics(0);
+
+	    // TODO Is this needed or set in RecordOffscreenCommandBuffer() ?
+            // Set the dynamic viewport and scissor
+            //commandBuffer.viewPortCmd(cvkSwapChain.GetWidth(), cvkSwapChain.GetHeight(), stack);
+            //commandBuffer.scissorCmd(cvkDevice.GetCurrentSurfaceExtent(), stack);
+            
+            // Check flags and render the nodes and connections
+            /// Loop through command buffers of hit test objects and record their buffers
+            hitTestRenderables.forEach(renderable -> {
+            // TODO HYDRA - WIP HIT TESTER
+//                if (renderable.GetVertexCount() > 0) {
+//                    renderable.RecordOffscreenCommandBuffer(inheritanceInfo, 0);
+//                    vkCmdExecuteCommands(commandBuffer.GetVKCommandBuffer(), renderable.GetCommandBuffer(0));
+//                }
+            });
+            
+            vkCmdEndRenderPass(commandBuffer.GetVKCommandBuffer());
+        
+            // Pre Draw Barrier
+            commandBuffer.pipelineImageMemoryBarrierCmd(cvkImage.GetImageHandle(), 
+                    VK_IMAGE_LAYOUT_UNDEFINED, VK_IMAGE_LAYOUT_GENERAL,//VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL, VK_IMAGE_LAYOUT_GENERAL,    // Old/New Layout
+                    0, VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT,                                // Src/Dst Access mask
+                    VK_PIPELINE_STAGE_ALL_COMMANDS_BIT, VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT, //VK_PIPELINE_STAGE_ALL_COMMANDS_BIT, VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT,  // Src/Dst Stage mask
+                    0, 1);                                                                  // baseMipLevel/mipLevelCount
+            
+            // TODO Do we need this?
+            vkDeviceWaitIdle(cvkDevice.GetDevice());
```

Comment:
Not sure but cvkDevice has a WaitIdle function.

Reply:
Still WIP so will figure it out